python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -268,24 +268,38 @@ def get_asset_materializations(
     before_timestamp: Optional[float] = None,
     after_timestamp: Optional[float] = None,
     tags: Optional[Mapping[str, str]] = None,
+    storage_ids: Optional[Sequence[int]] = None,
 ) -> Sequence[EventLogEntry]:
     check.inst_param(asset_key, "asset_key", AssetKey)
     check.opt_int_param(limit, "limit")
     check.opt_float_param(before_timestamp, "before_timestamp")
     check.opt_mapping_param(tags, "tags", key_type=str, value_type=str)
 
     instance = graphene_info.context.instance
-    event_records = instance.get_event_records(
-        EventRecordsFilter(
-            event_type=DagsterEventType.ASSET_MATERIALIZATION,
-            asset_key=asset_key,
-            asset_partitions=partitions,
-            before_timestamp=before_timestamp,
-            after_timestamp=after_timestamp,
-            tags=tags,
-        ),
-        limit=limit,
+    records_filter = AssetRecordsFilter(
+        asset_key=asset_key,
+        asset_partitions=partitions,
+        before_timestamp=before_timestamp,
+        after_timestamp=after_timestamp,
+        tags=tags,
```

Comment:
interestingly i am finding that taking these out and *just* passing in the storage IDs when you have a list of storage IDs already makes it return faster?? Which might make some sense since the other fields are redundant once you have storage IDs and might just make the query more complicated

Reply:
in our test harness i am seeing that query return in ~20 seconds once we take those additional filters out